yt_dlp/extractor/tunein.py

```diff
@@ -1,244 +1,332 @@
+            'thumbnail': r're:https?://.+',
+        },
+        'params': {'skip_download': 'Livestream'},
+    }]
+
+    def _real_extract(self, url):
+        embed_id = self._match_id(url)
+        kind = {
+            'p': 'program',
+            's': 'station',
+            't': 'topic',
+        }.get(embed_id[:1])
+
+        return self.url_result(
+            f'https://tunein.com/{kind}/?{kind}id={embed_id[1:]}')
+
+
 class TuneInShortenerIE(InfoExtractor):
-    _WORKING = False
     IE_NAME = 'tunein:shortener'
     IE_DESC = False  # Do not list
-    _VALID_URL = r'https?://tun\.in/(?P<id>[A-Za-z0-9]+)'
-
+    _VALID_URL = r'https?://tun\.in/(?P<id>[^/?#]+)'
     _TESTS = [{
-        # test redirection
         'url': 'http://tun.in/ser7s',
         'info_dict': {
             'id': 's34682',
             'title': str,
+            'ext': 'mp3',
+            'alt_title': 'World Class Jazz',
+            'channel_follower_count': int,
             'description': 'md5:d6d0b89063fd68d529fa7058ee98619b',
-            'thumbnail': r're:https?://cdn-profiles\.tunein\.com/.+',
             'location': 'Seattle-Tacoma, US',
-            'ext': 'mp3',
             'live_status': 'is_live',
+            'thumbnail': r're:https?://.+',
+        },
+        'params': {'skip_download': 'Livestream'},
+    }, {
+        'url': 'http://tun.in/tqeeFw',
+        'info_dict': {
+            'id': 't236404354',
+            'title': str,
+            'ext': 'mp3',
+            'alt_title': 'Technology Podcasts >',
+            'cast': 'count:1',
+            'description': 'md5:1029895354ef073ff00f20b82eb6eb71',
+            'display_id': '236404354',
+            'duration': 8330,
+            'series': 'Lex Fridman Podcast',
+            'series_id': 'p1153019',
+            'thumbnail': r're:https?://.+',
+            'timestamp': 1673458571,
+            'upload_date': '20230111',
         },
-        'params': {
-            'skip_download': True,  # live stream
+        'params': {'skip_download': 'Livestream'},
+    }, {
+        'url': 'http://tun.in/pei6i',
+        'info_dict': {
+            'id': 'p14',
+            'title': 'BBC News',
         },
+        'playlist_mincount': 35,
     }]
 
     def _real_extract(self, url):
         redirect_id = self._match_id(url)
         # The server doesn't support HEAD requests
-        urlh = self._request_webpage(
-            url, redirect_id, note='Downloading redirect page')
-
-        url = urlh.url
-        url_parsed = urllib.parse.urlparse(url)
-        if url_parsed.port == 443:
-            url = url_parsed._replace(netloc=url_parsed.hostname).url
+        urlh = self._request_webpage(url, redirect_id, 'Downloading redirect page')
+        parsed = urllib.parse.urlparse(urlh.url)
 
-        self.to_screen(f'Following redirect: {url}')
-        return self.url_result(url)
+        return self.url_result(
+            urllib.parse.urlunparse(parsed._replace(netloc=parsed.hostname)))
```

Comment:
```suggestion
        new_url = urllib.parse.urlunparse(parsed._replace(netloc=parsed.hostname))
        if self.suitable(new_url):  # Prevent infinite loop in case redirect fails
            raise UnsupportedError(new_url)
        return self.url_result(new_url)
```